tools/witx/src/docs/ast.rs

```diff
@@ -0,0 +1,487 @@
+                node.content_ref_mut::<MdNamedType>().r#type = Some(MdType::Alias {
+                    r#type: n.name.as_str().to_owned(),
+                })
+            }
+        }
+    }
+}
+
+impl ToMarkdown for NamedType {
+    fn generate(&self, node: MdNodeRef) {
+        self.tref.generate(node.clone());
+    }
+}
+
+impl ToMarkdown for Type {
+    fn generate(&self, node: MdNodeRef) {
+        match self {
+            Self::Enum(a) => a.generate(node.clone()),
+            Self::Int(a) => a.generate(node.clone()),
+            Self::Flags(a) => a.generate(node.clone()),
+            Self::Struct(a) => a.generate(node.clone()),
+            Self::Union(a) => a.generate(node.clone()),
+            Self::Handle(a) => a.generate(node.clone()),
+            Self::Array(a) => {
+                node.content_ref_mut::<MdNamedType>().r#type = Some(MdType::Array {
+                    r#type: a.type_name().to_owned(),
+                })
+            }
+            Self::Pointer(a) => {
+                node.content_ref_mut::<MdNamedType>().r#type = Some(MdType::Pointer {
+                    r#type: a.type_name().to_owned(),
+                })
+            }
+            Self::ConstPointer(a) => {
+                node.content_ref_mut::<MdNamedType>().r#type = Some(MdType::ConstPointer {
+                    r#type: a.type_name().to_owned(),
+                })
+            }
+            Self::Builtin(a) => {
+                node.content_ref_mut::<MdNamedType>().r#type = Some(MdType::Builtin {
+                    repr: a.type_name().to_owned(),
+                })
+            }
+        }
+    }
+}
+
+impl ToMarkdown for EnumDatatype {
+    fn generate(&self, node: MdNodeRef) {
+        let header = "#".repeat(node.borrow().ancestors().len() + 1);
+        node.new_child(MdSection::new(header.as_str(), "Variants"));
+
+        for variant in &self.variants {
+            let name = variant.name.as_str();
+            let id = if let Some(id) = node.any_ref().id() {
+                format!("{}.{}", id, name)
+            } else {
+                name.to_owned()
+            };
+            node.new_child(MdNamedType::new("-", id.as_str(), name, &variant.docs));
+        }
+
+        node.content_ref_mut::<MdNamedType>().r#type = Some(MdType::Enum {
+            repr: self.repr.type_name().to_owned(),
+        });
+    }
+}
+
+impl ToMarkdown for IntDatatype {
+    fn generate(&self, node: MdNodeRef) {
+        let header = "#".repeat(node.borrow().ancestors().len() + 1);
+        node.new_child(MdSection::new(header.as_str(), "Consts"));
+
+        for r#const in &self.consts {
+            let name = r#const.name.as_str();
+            let id = if let Some(id) = node.any_ref().id() {
+                format!("{}.{}", id, name)
+            } else {
+                name.to_owned()
+            };
+            let tt = MdNamedType::new("-", id.as_str(), name, &r#const.docs);
+            // TODO handle r#const.value
```

Comment:
Sounds good. I can add such a field but I'd like to keep a `TODO` there to remind ourselves to verify whether a better approach could potentially be found.

Reply:
On second thought, could we leave this one out until I add some tests and rethink `MdType` to be more recursive in nature? Otherwise, I worry this would not be correct. What do you think?

Reply:
Works for me